### PR TITLE
test(all): integration tests for Taiko network

### PR DESCRIPTION
**Pull request by johntaiko**

### CODE REVIEW

File: hive.go [hunk 1]
```diff
@@ -111,6 +111,9 @@ func main() {
 	runner := libhive.NewRunner(inv, builder, cb)
 	clientList := splitAndTrim(*clients, ",")
 
+	clientList = addTaiko(clientList)
```
Comment (alexshliu) [inline hive.go]: Do not modify here, use 
`./hive --sim taiko/rpc --client=taiko-l1,taiko-geth,taiko-driver,taiko-proposer,taiko-protocol,taiko-prover --docker.output --docker.pull=true`
 may be better? This is good for merging updates from ethereum/hive。

File: internal/libdocker/builder.go [hunk 2]
```diff
@@ -57,15 +58,26 @@ func (b *Builder) BuildClientImage(ctx context.Context, name string) (string, er
...
 	tag := fmt.Sprintf("hive/clients/%s:latest", name)
-	err := b.buildImage(ctx, dir, branch, tag)
+	err := b.buildImage(ctx, dir, "Dockerfile", branch, tag)
 	return tag, err
 }
 
 // BuildSimulatorImage builds a docker image of a simulator.
 func (b *Builder) BuildSimulatorImage(ctx context.Context, name string) (string, error) {
 	dir := b.config.Inventory.SimulatorDirectory(name)
+	buildContextPath := dir
```
Comment (alexshliu) [inline internal/libdocker/builder.go:68]: The changes in this file come from optimism? Do we need to add a copyright notice

File: scripts/gen_bindings.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
Comment (alexshliu) [inline scripts/gen_bindings.sh:1]: where is this file used？

File: simulators/eth2/engine/chain_generators.go [hunk 4]
```diff
@@ -56,7 +56,7 @@ func (p *PoWChainGenerator) Generate(genesis *setup.Eth1Genesis) ([]*types.Block
 	db := rawdb.NewMemoryDatabase()
 	engine := ethash.New(p.Config, nil, false)
 	insta := instaSeal{engine}
-	genesisBlock := genesis.Genesis.ToBlock(db)
+	genesisBlock := genesis.Genesis.ToBlock()
```
Comment (alexshliu) [inline simulators/eth2/engine/chain_generators.go:59]: why modify here?

File: simulators/taiko/rpc/go.mod [hunk 5]
```diff
@@ -0,0 +1,68 @@
+module github.com/ethereum/hive/simulators/taiko/rpc
+
+go 1.18
```
Comment (alexshliu) [inline simulators/taiko/rpc/go.mod:3]: should we use 1.18 or 1.19?

File: simulators/taiko/rpc/main.go [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+)
+
+type testSpec struct {
+	Name  string
+	About string
+	Run   func(*LegacyTestEnv)
+}
+
+var tests = []testSpec{
+	{Name: "http/BalanceAndNonceAt", Run: balanceAndNonceAtTest},
```
Comment (alexshliu) [inline simulators/taiko/rpc/main.go:21]: Do we need these tests for taiko-geth?

File: taiko/node.go [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+)
+
+type Node struct {
+	*hivesim.Client
+	role      string
+	opts      []hivesim.StartOption
+	TaikoAddr common.Address
+}
+
+type NodeOption func(*Node) *Node
```
Comment (johntaiko) [inline taiko/node.go:24]: Function optional maybe doesn't need return the `*Node`

Comment (alexshliu) [reply taiko/node.go:24]: I will refactor this later. This is designed to be  used for chain calling  like opt2(opt2(node)).

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**